pincer/middleware/interaction_create.py

```diff
@@ -56,23 +64,59 @@ async def interaction_create_middleware(self, payload: GatewayDispatch):
         if should_pass_cls(command.call):
             kwargs["self"] = self
 
-        message = await command.call(**kwargs)
+        if isasyncgenfunction(command.call):
+            message = command.call(**kwargs)
+            started = False
 
-        if isinstance(message, Embed):
-            message = Message(embeds=[message])
-        elif not isinstance(message, Message):
-            message = Message(message) if message else Message(
-                self.received_message,
-                flags=InteractionFlags.EPHEMERAL
-            )
+            async for msg in message:
 
-        await self.http.post(
-            f"interactions/{interaction.id}/{interaction.token}/callback",
-            message.to_dict()
-        )
+                msg = convert_message(self, msg)
+                if started:
+                    try:
+                        await self.http.post(
+                            f"webhooks/{interaction.application_id}/{interaction.token}",
+                            msg.to_dict().get("data")
+                        )
+                    except RateLimitError as e:
+                        _log.exception(
+                            f"RateLimitError: {e}. Retrying in {e.json.get('retry_after', 40)} seconds")
+                        await sleep(e.json.get("retry_after", 40))
+                        await self.http.post(
+                            f"webhooks/{interaction.application_id}/{interaction.token}",
+                            msg.to_dict().get("data")
+                        )
+
+                else:
+                    started = True
+
+                    await self.http.post(
+                        f"interactions/{interaction.id}/{interaction.token}/callback",
+                        msg.to_dict()
+                    )
+                await sleep(0.3)
```

Comment:
why `await sleep(.3)`?

Reply:
The delay is needed otherwise there will be a RateLimitError. `0.3` is the "sweetspot" as `0.25` does not always help and `0.5` is too slow.